src/foolscap/slicer.py

```diff
@@ -129,11 +131,22 @@ def slicerForObject(self, obj):
 BananaUnslicerRegistry = {}
 
 def registerUnslicer(opentype, factory, registry=None):
+
     if registry is None:
         registry = UnslicerRegistry
-    assert not registry.has_key(opentype)
+    assert opentype not in registry
+
+
     registry[opentype] = factory
 
+    # PY3KPORT  
+    if type(opentype) is tuple:
+        key = (six.ensure_binary(opentype[0]),)
```

Comment:
`isinstance` is generally preferred...but also what if this is more than a 1-tuple?